frontend/src/components/maps/leaflet/InfoSlideOut/InfoSlideOut.tsx

```diff
@@ -98,13 +107,72 @@ export type InfoControlProps = {
  */
 const InfoControl: React.FC<InfoControlProps> = ({ open, setOpen, onHeaderActionClick }) => {
   const popUpContext = React.useContext(PropertyPopUpContext);
+  const leaflet = useLeaflet();
+  const propertyInfo = popUpContext.propertyInfo;
+  const location = useLocation();
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const { getParcel } = useApi();
+  const jumpToView = () =>
+    leaflet.map?.setView(
+      [propertyInfo?.latitude as number, propertyInfo?.longitude as number],
+      MAX_ZOOM,
+    );
+  const zoomToView = () =>
+    leaflet.map?.flyTo(
+      [propertyInfo?.latitude as number, propertyInfo?.longitude as number],
+      MAX_ZOOM,
+    );
 
   useEffect(() => {
     const elem = L.DomUtil.get('infoContainer');
     if (elem) {
       L.DomEvent.on(elem!, 'mousewheel', L.DomEvent.stopPropagation);
     }
   });
+  const addAssociatedBuildingLink = (
+    <>
+      <FaPlusSquare color="#1a5a96" className="mr-1" />
+      <Link
+        to={{
+          pathname: `/mapview`,
+          search: queryString.stringify({
+            ...queryString.parse(location.search),
+          }),
+        }}
+        onClick={() => {
+          toast.dark('Creating new associated building...');
+          getParcel(propertyInfo?.id as number)
+            .then((parcel: IParcel) => {
+              jumpToView();
+              dispatch(
+                storeAssociatedBuilding({
+                  ...defaultBuildingValues,
+                  parcelId: parcel?.id,
+                  parcels: [{ ...parcel }],
+                }),
+              );
```

Comment:
is there anyway this could lead to duplicate buildings being created? (like if someone wants to attach an existing building but clicks the link on the slide out and creates a new one instead) or do we even care/check for that in general?

Reply:
It would create a duplicate if they went all the way through the inventory form and saved it. However, theoretically when the looked at the map they would see the pin on the building already and wouldn't add it again. Also, eventually we'll be showing all of the existing associated buildings on the slideout, so they'll see their building already exists and hopefully won't feel the need to add it again.